src/integritysetup/integrity_common.c

```
@@ -0,0 +1,58 @@
+#include "parse-util.h"
+
+int parse_integrity_options(const char *options, uint32_t *activate_flags, struct crypt_params_integrity *p, char **data_device, char **integrity_algr) {
+        int r;
+
+        for (;;) {
+                _cleanup_free_ char *word = NULL;
+                char *val;
+
+                r = extract_first_word(&options, &word, ",", EXTRACT_DONT_COALESCE_SEPARATORS | EXTRACT_UNESCAPE_SEPARATORS);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to parse options: %m");
+                if (r == 0)
+                        break;
+                if (isempty(word))
+                        continue;
+                else if (streq(word, "allow-discards")) {
+                        if (activate_flags)
+                                *activate_flags |= CRYPT_ACTIVATE_ALLOW_DISCARDS;
+                } else if ((val = startswith(word, "journal-watermark="))) {
+                        uint32_t percent;
+                        r = safe_atou32(val, &percent);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to parse journal-watermark value (%s)", val);
+
+                        if (percent > 100) {
+                                log_warning("journal-watermark domain is 0..100, using default for invalid value %u\n", percent);
+                        } else {
+                                if (p)
+                                        p->journal_watermark = percent;
+                        }
+                } else if ((val = startswith(word, "journal-commit-time="))) {
+                        uint32_t commit_time;
+                        r = safe_atou32(val, &commit_time);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to parse journal-commit-time value (%s)", val);
+                        if (p)
+                                p->journal_commit_time = commit_time;
```

hmm what happens if this is set to zero?

---

This results in using the default.  This is true for percentage as well.  I'll update the man page.   I suppose we could return an error if 0 is specified for either of these two optional options, so that the user is required to remove the option of they want the default?

---

I updated man page.